src/server/enterprise/testing/cmd_test.go

```diff
@@ -234,3 +235,83 @@ func TestLoginPachd(t *testing.T) {
 		"user", tu.DexMockConnectorEmail,
 	).Run())
 }
+
+// Tests synching contexts from the enterprise server
+func TestSyncContexts(t *testing.T) {
+	resetClusterState(t)
+	defer resetClusterState(t)
+
+	id := tu.UniqueString("cluster")
+	clusterId := tu.UniqueString("clusterDeploymentId")
+
+	// register a new cluster
+	require.NoError(t, tu.BashCmd(`
+		echo {{.license}} | pachctl enterprise activate
+		echo {{.token}} | pachctl auth activate --enterprise --issuer http://pach-enterprise.enterprise:658 --supply-root-token
+		pachctl enterprise register --id {{.id}} --enterprise-server-address grpc://pach-enterprise.enterprise:650 --pachd-address grpc://pachd.default:650 --pachd-user-address grpc://pachd.default:655 --cluster-deployment-id {{.clusterId}} 
+		`,
+		"id", id,
+		"token", tu.RootToken,
+		"license", tu.GetTestEnterpriseCode(t),
+		"clusterId", clusterId,
+	).Run())
+
+	// assert the registered cluster isn't reflected in the user's config
+	require.YesError(t, tu.BashCmd(`
+		pachctl config list context | match {{.id}}
+		`,
+		"id", id,
+	).Run())
+
+	// sync contexts and assert that the newly registered cluster is accessible
+	require.NoError(t, tu.BashCmd(`
+		pachctl enterprise sync-contexts
+		pachctl config list context | match {{.id}}
+		pachctl config get context {{.id}} | grep "\"pachd_address\": \"grpc://pachd.default:655\""
+		pachctl config get context {{.id}} | grep "\"cluster_deployment_id\": \"{{.clusterId}}\""
+		pachctl config get context {{.id}} | grep "\"source\": \"IMPORTED\","
+		`,
+		"id", id,
+		"clusterId", clusterId,
+	).Run())
+
+}
+
+// Tests RegisterCluster command's derived argument values if not provided
+func TestRegisterDefaultArgs(t *testing.T) {
+	resetClusterState(t)
+	defer resetClusterState(t)
+
+	resetClusterState(t)
+	defer resetClusterState(t)
+
+	c, err := client.NewForTest()
+	require.NoError(t, err)
+
+	id := tu.UniqueString("cluster")
+
+	// get cluster ID from connection
+	clusterInfo, inspectErr := c.AdminAPIClient.InspectCluster(c.Ctx(), &types.Empty{})
+	require.NoError(t, inspectErr)
+	clusterId := clusterInfo.DeploymentID
+
+	// register a new cluster
+	require.NoError(t, tu.BashCmd(`
+		echo {{.license}} | pachctl enterprise activate
+		echo {{.token}} | pachctl auth activate --enterprise --issuer http://pach-enterprise.enterprise:658 --supply-root-token
+		pachctl enterprise register --id {{.id}} --enterprise-server-address grpc://pach-enterprise.enterprise:650 --pachd-address grpc://pachd.default:650
```

Comment:
@actgardner the one issue here is that I'm not testing leaving pachd-address argument blank. If I were to do so, the enterprise server will try to look for pachd at an address inaccessible to it. 

Do you think there's a good way to test leaving pachd-address blank?

Reply:
It doesn't feel great to say "don't test it" but we might need to skip it for now - I don't have any great ideas for how to handle this with our current black box approach to testing.